clang/test/Analysis/Checkers/WebKit/unretained-call-args.mm

```diff
@@ -438,6 +438,20 @@ void use_const_local() {
 
 } // namespace const_global
 
+namespace ns_retained_return_value {
+
+NSString *provideNS() NS_RETURNS_RETAINED;
+CFDictionaryRef provideCF() CF_RETURNS_RETAINED;
+void consumeNS(NSString *);
+void consumeCF(CFDictionaryRef);
+
+void foo() {
+  consumeNS(provideNS());
+  consumeCF(provideCF());
+}
+
+} // namespace ns_retained_return_value
+
```

Comment:
I'm fairly certain this works, but can we have a test case with inheritance (sub class calls `provideNS` defined in super class) for the sake of completion?

Reply:
You mean `provideNS` returns a subclass of a class `consumeNS` takes?